bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/SolaxBridgeHandler.java

```diff
@@ -0,0 +1,135 @@
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolaxBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Konstantin Polihronov - Initial contribution
+ */
+@NonNullByDefault
+public class SolaxBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SolaxBridgeHandler.class);
+    private static final int INITIAL_SCHEDULE_DELAY_SECONDS = 5;
+
+    private SolaxConfiguration config;
+
+    private LocalHttpConnector localHttpConnector;
+    private Set<ScheduledFuture<?>> schedules = new HashSet<>();
+
+    public SolaxBridgeHandler(Bridge bridge) {
+        super(bridge);
+
+        config = getConfigAs(SolaxConfiguration.class);
+        localHttpConnector = new LocalHttpConnector(config.password, config.hostname);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initialize()...");
+        updateStatus(ThingStatus.UNKNOWN);
+
+        int refreshInterval = config.refresh;
+        TimeUnit timeUnit = TimeUnit.SECONDS;
+        logger.debug("Scheduling regular interval retrival on every {} {}", refreshInterval, timeUnit);
+        ScheduledFuture<?> fixedDelaySchedule = scheduler.scheduleWithFixedDelay(this::retrieveData,
+                INITIAL_SCHEDULE_DELAY_SECONDS, refreshInterval, timeUnit);
+        schedules.add(fixedDelaySchedule);
+    }
+
+    private void retrieveData() {
+        try {
+            String rawJsonData = localHttpConnector.retrieveData();
+            logger.debug("Raw data retrieved = {}", rawJsonData);
+
+            updateInverterData(rawJsonData);
+
+            Bridge bridge = getThing();
+            if (bridge.getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            logger.warn("Exception received while attempting to retrieve data via HTTP", e);
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    private void updateInverterData(@Nullable String rawJsonData) {
+        LocalConnectRawDataBean inverterParsedData = LocalConnectRawDataBean.fromJson(rawJsonData);
+        Bridge bridge = getThing();
+        List<Thing> things = bridge.getThings();
+        if (inverterParsedData != null) {
+            for (Thing thing : things) {
+                ThingHandler handler = thing.getHandler();
+                if (handler instanceof InverterDataUpdateListener listener) {
+                    listener.updateListener(inverterParsedData);
```

Comment:
It seems some filtering is needed here as otherwise all inverters will receive data for all inverters, and they are also not filtering, so all channels will be overwritten all the time?

Reply:
That's maybe a bit of a modelling issue because currently it's 1:1 bridge / inverter. The idea is the inverter to be the "data" and the bridge to be the way to retrieve the data. Currently this retrieves the data from the local wifi module but the next improvement will be to also retrieve the same data from the cloud (for wifi modules version <3.x). Then the same data will be pushed to the inverter but using another bridge. If you have some suggestion how to handle this model please let me know...

Reply:
In that case I would advise to remove the Bridge thing and simply model the combined inverter+Wi-Fi module as one Thing. Bridges should be used to model physical or logical bridges/gateways, but this seems not to be the case here.

Examples:
- **DanfossAirUnit:** The communication with the unit is done through a connected Z-Wave module. There is no need for a bridge since the communication module can be considered a part of the unit itself making communication possible.
- **Miele@home:** Each appliance (e.g. washing machine) has a Zigbee or Wi-Fi communication module, or has built-in Wi-Fi. The API is exposed by a Zigbee gateway (including Wi-Fi devices) which is connected to the LAN. There is no need for a gateway representing each appliance/communication module, only a Bridge representing the Zigbee gateway. In fact the binding cannot know if an appliance has built-in Wi-Fi or a plugged in Wi-Fi module.
- **Netatmo:** Here we have two layers of Bridges: The cloud API itself is a Bridge, since you have have multiple weather stations connected to the same Netatmo account. Each weather station is also a bridge since indoor/outdoor modules communicates through it by RF.

When you will implement the cloud service you could create another Thing for that. You can still reuse the code, but you cannot migrate the same Thing between cloud and local. Is there any need to do that? I would suppose that users having the Wi-Fi module would prefer to integrate locally. So the usual migration path would probably be cloud -> local. In that case one could simply create a new Thing and relink the channels.

The two integrations might also deviate at some point anyway, for example some channels might be available locally, but not through cloud, or vice versa.

Reply:
Hm.... that definitely makes sense but may require severe refactoring (including the documentation). I didn't know that we can create directly things that also do the communication part. Thought that any communication related "thing" should be a bridge.
I'll leave this topic last for a review, because this Friday I go on vacation and will not be available the next 10 days or so...

Reply:
OK. I think I managed to do that in the code and also tested it. Looks to be working (at least like before) :)

Reply:
@jlaur if we have only one thing that connects to an IP address/hostname and retrieves all the data, then the discovery service becomes useless, right? I don't see why would we want to use it....

Reply:
Yes. However, if you can discover the Wi-Fi module using mDNS or UPnP, that would still be useful.

Reply:
I think it's not possible. Will remove the whole class for now...